### ISSUE TITLE
React.renderComponentToString() and ExecutionEnvironment.canUseDOM

### ISSUE THREAD

**Issue by insin** [facebook/react#1129]
I'm using `React.renderComponentToString()` to test generated output from my [forms library](https://github.com/insin/newforms/tree/react), sharing the same set of QUnit tests between Firefox, Chrome and Node.js (using [qqunit](https://github.com/hornairs/qqunit)).

This function has always thrown errors for me when used in the browser, but my tests still worked because it seemed to call the callback before throwing the error in the [custom assertion function](https://gist.github.com/insin/9005207) I've been using.

Trying 0.9.0-rc1:
- my browser test suites are complaining because the function is now synchronous so the error is being thrown before I can get a hold of generated HTML.
- when run in Node.js, the tests are bombing out on the navigator check for Chrome becuase qqunit fakes out a `window` global using jsdom, but not a `navigator`. If I patch in a check for navigator not being undefined, I then get the same results as when running in a browser.

So I edited the node_modules version of 0.9.0-rc1 to add a process check to `canUseDOM`:

``` javascript
var canUseDOM = typeof window !== 'undefined' && typeof process === 'undefined';
```

Now all my tests run in Node.js, but it complains about some assertions (because boolean HTML attributes no longer have the `="boolean"` part in the generated markup - no biggie).

I then edited the version of 0.9.0-rc1 my test browsers were using with the following:

``` javascript
var canUseDOM = false;
```

Now there are no more errors being thrown when I call `React.renderComponentToString()` and I have the same errors about HTML differences as I was getting when running in Node.js.

TL;DR: should React always be using the equivalent of `var canUseDOM = false;` when executing `React.renderComponentToString()`? It stops it from throwing errors.
